docs/src/weakdeps.md

```diff
@@ -0,0 +1,77 @@
+# Weak dependencies
+
+It is sometimes desirable to be able to extend some functionality of a package without having to
+unconditionally take on the cost (in terms of e.g. load time) of adding an extra dependency.
+A *weak* dependency is a package that is only available to load if some other package in the
+current environment have that package as a normal (or strong) dependency. 
+
+Weak dependencies are listed in a `Project.toml` file under the `[weakdeps]` section which can be compared to a
+(strong) dependency which is under the `[deps]` section.
+Compatibility on weak dependencies are specified like a normal dependeny unde the ´[compat]` section.
```

Comment:
```suggestion
Compatibility on weak dependencies are specified like a normal dependency under the `[compat]` section.
```

Reply:
Thanks, the current docs are still very WIP.